api.py

```diff
@@ -1,11 +1,97 @@
+import requests
+import uvicorn
+from fastapi import Body, FastAPI, Response
+from requests import RequestException
+from structlog import get_logger
+
+from app.validators.questionnaire_validator import QuestionnaireValidator
+
+app = FastAPI()
+
+
+@app.get("/status")
+async def status():
+    return Response(status_code=200)
+
+
+logger = get_logger()
+
+AJV_HOST = os.getenv("AJV_HOST", "localhost")
+
+AJV_VALIDATOR_URL = f"http://{AJV_HOST}:5002/validate"
+
+
+@app.post("/validate")
+async def validate_schema_request_body(payload=Body(None)):
+    logger.info("Validating schema")
+    return await validate_schema(payload)
+
+
+@app.get("/validate")
+async def validate_schema_from_url(url: str = ""):
+    if url:
+        logger.info("Validating schema from URL", url=url)
+        try:
+            with urllib.request.urlopen(url) as opened_url:
+                return await validate_schema(data=opened_url.read().decode())
+        except urllib.error.URLError:
+            return (
+                Response(
+                    status_code=404,
+                    content=f"Could not load schema at URL [{url}]",
+                ),
+            )
+
+
+async def validate_schema(data):
+    json_to_validate = None
+    if data:
+        if isinstance(data, str):
+            try:
+                json_to_validate = json.loads(data)
+            except JSONDecodeError:
+                logger.info("Could not parse JSON", status=400)
+                return Response(status_code=400, content="Could not parse JSON")
+        elif isinstance(data, dict):
+            json_to_validate = data
+
+    response = {}
+    try:
+        ajv_response = requests.post(
+            AJV_VALIDATOR_URL, json=json_to_validate, timeout=10
+        )
+        ajv_response_dict = ajv_response.json()
+
+        if ajv_response_dict:
+            response["errors"] = ajv_response_dict["errors"]
+            logger.info("Schema validator returned errors", status=400)
+            return response, 400
+
+    except RequestException:
+        logger.info("AJV Schema validator service unavailable")
+        return json.dumps(obj={}, error="AJV Schema validator service unavailable")
+
+    validator = QuestionnaireValidator(json_to_validate)
+    validator.validate()
+
+    if len(validator.errors) > 0:
+        response["errors"] = validator.errors
+        logger.info("Questionnaire validator returned errors", status=400)
+        response = Response(content=json.dumps(response), status_code=400)
+
+        return response
+
+    logger.info("Schema validation passed", status=200)
+
+    response = Response(content=json.dumps(response), status_code=200)
+
+    return response
 
-application = Flask(__name__)
-application.register_blueprint(validate_blueprint)
-application.register_blueprint(status_blueprint)
 
 if __name__ == "__main__":
-    application.run(port=5001)
+    uvicorn.run("api:app", workers=20, port=5001, reload=True)
```

Comment:
I get the below warning message when I run `make run`, due of the `reload` flag I think? 

```
WARNING:  "workers" flag is ignored when reloading is enabled.
```

Reply:
So I guess I will remove it, the "reload".

Reply:
Yeah , it's only for our local runs, so I guess we don't want multiple workers as much as reloads. It is easily adjustable in runner and validator anyway, just depends what do we want more? Fast validation or server auto reloading on changes.

Reply:
I guess we only should validate the schemas affected when testing locally so the speed of extra workers is not essential, I will keep reloads on.

Reply:
Yeah that makes sense 👍  My vote goes to 'reloads' but see what others think?

Reply:
I would call this out in readme, that if you want it running fast/workers locally need to set reload to False.

Reply:
Added to the README now, thanks.